docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
+- Horizontally scalable (capacity and performance)
+- Accessible from a central end point
+- Highly Available
+- Easy to use query language
+- Service accessible through a UI
+- Monitorable service (health, performances and alerts)
+- Ability to correlate logs with KPIs and alerts
+- Multitenancy for audit logs
```

Comment:
So, multi-tenancy means you want RBAC. Maybe you want different log sources, maybe you want to filter based on roles. But that's not going to be part of Metal's log aggregation system I think, multi-tenancy isn't really our concern, it's only at the application level. If they want multi-tenant S3 audit logs in cloudserver, this will probably need another design and shouldn't be aggregated in the central log collector.

Reply:
okay will remove everything about tenant audit logs

Reply:
RBAC and multi-tenancy are two completely different features to me. Probably RBAC is better addressed on the "viewing/display" instance rather than on the collection.
Same goes for Multi-tenancy, which is probably handled better somewhere else (higher in the solution, i.e. UI, CLI) rather than on the collection. Still (see above), I think we need to be able to collect audit-logs and for all logsg include the appropriate "hooks", so that they can be filtered appropriately to be only displayed to a specific tenant.